### ISSUE TITLE
Add Rate Limiting Support

### ISSUE THREAD

**Issue by thinkingserious** [sendgrid/sendgrid-php#325]
We would like to add support for v3 Web API rate limits, as described [here](https://sendgrid.com/docs/API_Reference/Web_API_v3/How_To_Use_The_Web_API_v3/rate_limits.html).

**_How would you like this to be implemented?_** Or please give a thumbs up to express your interest in us implementing to move this task up our queue.

PRs are always welcome as well :)

**Comment by budirec** [comment]
The easiest solution is to make an if statement in makeRequest to look for $statusCode = 429 then sleep until the rate limit refresh happen and just call the same function with counter to make sure we can get out from it

**Comment by thinkingserious** [comment]
For reference, [here](https://github.com/sendgrid/sendgrid-csharp/blob/master/USE_CASES.md#transient_faults) is the implementation in our C# SDK.

**Comment by jopanel** [comment]
Hi,

Based on the C# implementation I think this would be an easy case.


```
// Reliability settings measured in seconds.
$reliabilitySettings = [
"retryCount" => 2,
"minBackOff" => 1,
"maxBackOff" => 10,
"deltaBackOff" => 3
];

$email->reliabilitySettings($reliabilitySettings);
```

I can make a PR but I would consider adding it to the refactor PR.

**Comment by thinkingserious** [comment]
Hi @jopanel,

If you have time, it would be nice to get it in now, while we wait for the refactor PR to be completed. 

With Best Regards,

Elmer

**Comment by jopanel** [comment]
Hi

I hope I added it in the [right place for this PR](https://github.com/sendgrid/sendgrid-php/pull/453).

**Comment by thinkingserious** [comment]
Thanks @jopanel!

**Comment by mbernier** [comment]
@thinkingserious @jopanel I want to make sure I am following this properly, but is this issue closed by #439 ?
